app/bundles/CoreBundle/Controller/AbstractStandardFormController.php

```diff
@@ -108,42 +110,16 @@ protected function batchDeleteStandard(Request $request)
             ],
         ];
 
-        if ('POST' == $request->getMethod()) {
+        if (Request::METHOD_POST == $request->getMethod()) {
             $model     = $this->getModel($this->getModelName());
-            $ids       = json_decode($request->query->get('ids', ''));
-            $deleteIds = [];
-
-            // Loop over the IDs to perform access checks pre-delete
-            foreach ($ids as $objectId) {
-                $entity = $model->getEntity($objectId);
-
-                if (null === $entity) {
-                    $flashes[] = [
-                        'type'    => 'error',
-                        'msg'     => $this->getTranslatedString('error.notfound'),
-                        'msgVars' => ['%id%' => $objectId],
-                    ];
-                } elseif (!$this->checkActionPermission('batchDelete', $entity)) {
-                    $flashes[] = $this->accessDenied(true);
-                } elseif ($model->isLocked($entity)) {
-                    $flashes[] = $this->isLocked($postActionVars, $entity, $this->getModelName(), true);
-                } else {
-                    $deleteIds[] = $objectId;
-                }
-            }
-
-            // Delete everything we are able to
-            if (!empty($deleteIds)) {
-                $entities = $model->deleteEntities($deleteIds);
-
-                $flashes[] = [
-                    'type'    => 'notice',
-                    'msg'     => $this->getTranslatedString('notice.batch_deleted'),
-                    'msgVars' => [
-                        '%count%' => count($entities),
-                    ],
-                ];
-            }
+            $flashes   = $this->batchDeleteService->batchDelete(
+                $request,
+                $model,
+                $postActionVars,
+                $this->getSessionBase(),
+                $this->getModelName(),
+                [$this, 'isLocked'],
```

Comment:
I would create another service for this isLocked method but that can be done in the second step.